From 59216f5c8de05f18ef65f4285b4e9aad3cfb499d Mon Sep 17 00:00:00 2001
From: Alexander Stein <alexander.stein@ew.tq-group.com>
Date: Mon, 15 Jan 2024 08:05:34 +0100
Subject: [PATCH] iMX9: Use REV to select AHAB_IMG

Rev 1.0 (A0) and Rev 1.1 (A1) need different sentinel firmware files.
Hardware revision can be specified on commandline already, but AHAB_IMG
was not affected. Select firmware file based on REV variable.

Signed-off-by: Alexander Stein <alexander.stein@ew.tq-group.com>

---
 iMX9/soc.mak | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/iMX9/soc.mak b/iMX9/soc.mak
index 43d68e9..89e9452 100644
--- a/iMX9/soc.mak
+++ b/iMX9/soc.mak
@@ -16,7 +16,9 @@ else
     RENAME = rename
 endif
 
-AHAB_IMG = mx93a0-ahab-container.img
+LC_REVISION = $(shell echo $(REV) | tr ABC abc)
+
+AHAB_IMG = mx93$(LC_REVISION)-ahab-container.img
 MCU_IMG = m33_image.bin
 
 SPL_LOAD_ADDR ?= 0x2049A000
